cllc-public-app/Controllers/AdoxioLegalEntityController.cs

```diff
@@ -168,11 +167,17 @@ public IActionResult GetDynamicsLegalEntitiesByPosition(string parentLegalEntity
                 throw;
             }
 
+            string temp = _httpContextAccessor.HttpContext.Session.GetString("UserSettings");
+            UserSettings userSettings = JsonConvert.DeserializeObject<UserSettings>(temp);
 
             if (legalEntities != null)
             {
                 foreach (var legalEntity in legalEntities)
                 {
+                    // Users can't access other users legal entities.
+                    if(!AccountController.CurrentUserHasAccessToAccount(new Guid(legalEntity._adoxioAccountValue), _httpContextAccessor, _dynamicsClient)){
```

Comment:
Can you please move CurrentUserHasAccessToAccount to Dynamics Extensions - it's not a good pattern to reference a controller from another controller.  (Dynamics Extensions is currently in the Context folder)